src/status_im/wallet_connect_legacy/core.cljs

```diff
@@ -0,0 +1,188 @@
+                                                                :connector connector})
+        params (first (:params session))
+        metadata (:peerMeta params)
+        account (first (:accounts params))
+        sessions (get db :wallet-connect-legacy/sessions)
+        updated-sessions (if sessions (conj sessions session) [session])]
+    (log/debug "[wallet connect 1.0] session created - " session)
+    {:show-wallet-connect-success-sheet nil
+     :db (assoc db :wallet-connect/session-connected session :wallet-connect-legacy/sessions updated-sessions)}))
+
+(fx/defn manage-app
+  {:events [:wallet-connect-legacy/manage-app]}
+  [{:keys [db]} session]
+  {:db (assoc db :wallet-connect/session-managed session :wallet-connect/showing-app-management-sheet? true)
+   :show-wallet-connect-app-management-sheet nil})
+
+(fx/defn request-handler
+  {:events [:wallet-connect-legacy/request]}
+  [{:keys [db] :as cofx} request-event]
+  (let [request (js->clj request-event :keywordize-keys true)
+        params (:request request)
+        pending-requests (or (:wallet-connect-legacy/pending-requests db) [])
+        new-pending-requests (conj pending-requests request)
+        client (get db :wallet-connect-legacy/client)
+        topic (:topic request)]
+    {:db (assoc db :wallet-connect-legacy/pending-requests new-pending-requests)
+     :dispatch [:wallet-connect-legacy/request-received request]}))
+
+(fx/defn request-handler-test
+  {:events [:wallet-connect-legacy/request-test]}
+  [{:keys [db] :as cofx}]
+  {:show-wallet-connect-sheet nil})
+
+(fx/defn approve-proposal
+  {:events [:wallet-connect-legacy/approve-proposal]}
+  [{:keys [db]} account]
+  (let [connector (get db :wallet-connect-legacy/proposal-connector)
+        proposal-chain-id (get db :wallet-connect-legacy/proposal-chain-id)
+        address (ethereum/normalized-hex (:address account))
+        accounts [address]]
+    (^js .approveSession connector (clj->js {:accounts accounts :chainId proposal-chain-id}))
+    {:hide-wallet-connect-sheet nil}))
+
+(fx/defn reject-proposal
+  {:events [:wallet-connect-legacy/reject-proposal]}
+  [{:keys [db]} account]
+  (let [connector (get db :wallet-connect-legacy/proposal-connector)]
+    (^js .rejectSession connector)
+    {:hide-wallet-connect-sheet nil}))
+
+(fx/defn change-session-account
+  {:events [:wallet-connect-legacy/change-session-account]}
+  [{:keys [db]} session account]
+  (let [connector (:connector session)
+        address (:address account)]
+    (^js .updateSession connector (clj->js {:chainId 1 :accounts [address]}))
+    {:hide-wallet-connect-app-management-sheet nil
+     :db (assoc db :wallet-connect/showing-app-management-sheet? false)}))
+
+(fx/defn disconnect-session
+  {:events [:wallet-connect-legacy/disconnect]}
+  [{:keys [db]} session]
+  (let [sessions (get db :wallet-connect-legacy/sessions)
+        connector (:connector session)]
+    (^js .killSession connector)
```

Comment:
should be moved to effects. I know this changes not that much but maybe if someone will want to write tests or something, might be helpful to move those side effects